src/app/services/spell.service.ts

```diff
@@ -2,40 +2,81 @@ import { Injectable } from '@angular/core';
 import { HttpClient } from '@angular/common/http';
 import { Observable } from 'rxjs';
 import { map } from 'rxjs/operators';
+import { forkJoin } from 'rxjs';
 
-import { Spell, SummonSpell } from '../model/spell';
+import { Spell, ShapeshiftSpell, SummonSpell } from '../model/spell';
 
 @Injectable({
-    providedIn: 'root'
+  providedIn: 'root'
 })
 export class SpellService {
-    private spellsUrl = 'api/spells';
+  private spellsUrl = 'api/spells';
 
-    constructor(private http: HttpClient) {
+  constructor(private http: HttpClient) {
 
-    }
+  }
 
-    getSpells(): Observable<Spell[]> {
-        return this.http.get<Spell[]>(this.spellsUrl);
-    }
+  getSpells(): Observable<Spell[]> {
+    return this.http.get<Spell[]>(this.spellsUrl);
+  }
 
-    getSpellsByGroup(group: string): Observable<Spell[]> {
-        return this.http.get<Spell[]>(this.spellsUrl + '/?group=' + group);
-    }
 
-    getSpellsByCharacterGroupAndLevel(group: string, level: number): Observable<Spell[]> {
-        return this.http.get<Spell[]>((this.spellsUrl) + '/?group=' + group).pipe(
-            map(spells => {
-                return spells.filter(spell => spell.level <= level);
-            })
-        );
-    }
+  getSpellsByCharacterClassAndLevel(characterClass: string, level: number): Observable<Spell[]> {
+    let spells: Observable<Spell[]>;
+    const spellGroup = this.calculateSpellGroupFromClass(characterClass);
+
+    // TODO: How do I combine these two items into one returnable Observable?
```

Comment:
@rdrrichards I need help on this... I want to make one call to the Spell Service and get back a (observable) list of Spells, some of them Summon spells and some of them Shapeshift spells and some of them regular spells. I've been experimenting with forkjoin and not able to get it to do what I want. I'm probably missing something obvious.

Reply:
@cvillhauer Try this:

```
  getSpellsByCharacterClassAndLevel(characterClass: string, level: number): Observable<Spell[]> {
    const spellGroup = this.calculateSpellGroupFromClass(characterClass);

    // TODO: How do I combine these two items into one returnable Observable?
    return forkJoin([this.getShapeshiftSpellsBySpellLevel(level),
      this.getSummonSpellsBySpellGroupAndLevel(spellGroup, level)]).pipe(
        map(value => [...value[0], ...value[1]])
      );
  }
```